app/features/onboarding/onboarding_controller.rb

```diff
@@ -127,6 +127,18 @@ class OnboardingController < ControllerBase
     # 3) If the data is not valid, display an error message
     post "#{HOME}/initial_clone_user" do
       if valid_params?(params, post_parameter_list_for_clone_user_validation)
+        api_token = params[:clone_user_api_token]
+        api_token.strip!
+        if api_token.length != 40
+          error_message = "Invalid API token, please ensure you entered an API token and not a password"
+          session[:message] = error_message
+          logger.error(error_message)
+          logger.error("API tokens are generated by GitHub and are 40 characters long")
+
+          redirect("#{HOME}/initial_clone_user")
+          return
```

Comment:
Could you please remove the extra `return`?

Reply:
Does `redirect` cause the rest of the method to not be executed? I don't want the rest of the logic in this method to execute.

Reply:
Yep! 🙂 

```ruby
require "sinatra"

get "/" do
  puts "Before redirect."
  redirect "/redirect_here"
  puts "After redirect."
end

get "/redirect_here" do
  puts "Redirected here."
  "Hello World!"
end
```

```
❯ ruby test.rb                                                                                                                                                                                                                             
== Sinatra (v2.0.1) has taken the stage on 4567 for development with backup from Thin
Thin web server (v1.7.2 codename Bachmanity)
Maximum connections set to 1024
Listening on localhost:4567, CTRL+C to stop
Before redirect.
::1 - - [12/Apr/2018:18:52:09 -0400] "GET / HTTP/1.1" 302 - 0.0072
Redirected here.
::1 - - [12/Apr/2018:18:52:09 -0400] "GET /redirect_here HTTP/1.1" 200 12 0.0006
::1 - - [12/Apr/2018:18:52:09 -0400] "GET /favicon.ico HTTP/1.1" 404 469 0.0008
```

Reply:
That's magical 🌟

Reply:
🔮